lib/reports/eligible_commitments.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require "services"
+require "cluster"
+
+Services.mongo!
+
+module Reports
+  # Given criteria, pull up all holdings that match those criteria
+  # AND are eligible for commitments AND have no commitments.
+  # Invoke via bin/reports/compile_eligible_commitments_report.rb
+  class EligibleCommitments
+    def header
+      [
+        "organization",
+        "oclc_sym",
+        "ocn",
+        "local_id"
+      ]
+    end
+
+    def for_ocns(ocns = [])
+      if ocns.empty?
+        raise "No ocns given"
+      end
+
+      ocns.sort.uniq.each do |ocn|
+        cluster = Cluster.find_by(ocns: [ocn.to_i])
+
+        next if cluster.nil?
+        next if we_have_seen? cluster._id
+        next unless cluster.eligible_for_commitments?
+        next unless cluster.commitments.count.zero?
+
+        cluster.holdings.select(&:eligible_for_commitments?).each do |holding|
+          yield [
+            holding.organization,
```

Comment:
For purposes of this report, do we care what HTID or cluster this commitment would correspond to?

Reply:
I don't think so, since commitments are on title level, sorta. We used to have resolved_oclc for this, but that's out now. Maybe cluster id, but that only makes sense if they are stable-ish.